### ISSUE TITLE
[Issue] Whenever a new setting is added, need to fix settings manually

### ISSUE THREAD

**Issue by Brazuca87** [NoxxDev/NecroBot#891]
### Expected Behavior
When a new setting is implemented, bot should add the missing setting with default values (maybe output an info saying settings were added), like it always did before.

### Actual Behavior
Bot complains that settings are missing, and need to fix manually.
[16:15:12] () config.json [Line: 60, Position: 27]: HumanWalkSnipeConfig Required properties are missing from object: UseFastPokemap, UsePogoLocationFeeder.

### Steps to Reproduce
Remove a line from config.json

### Other Information
Version: latest commit
Windows Version: 10
What is affected (catch, transfer, evolve, recycle, snipe, powerup/levelup, other): startup

**Comment by Brazuca87** [comment]
Another example:
```
[16:23:43] config.json [Line: 1, Position: 1]:  Required properties are missing from object: PokemonUpgradeFilters.
[16:23:43] Fix config.json and restart NecroBot or press a key to ignore and continue...
[16:23:46] Exception caught, writing LogBuffer.
```

**Comment by jjskuld** [comment]
I agree it's kind of broken especially when working off master branch.  I have a proposal to fix this and improve settings in general.

1. Upgrading config settings should be somewhat independent of the auto update functionality.  This way we can upgrade settings when developing off master.  Not should check for new settings on startup and upgrade old settings.

2. Default configuration should be written to config-defaults.json or something like that.  It should be regenerated on startup and written to file every time.  So people can use this as a template if they want.

3. We should generate a hash code of the default configuration and write that out as a version or version hash field in the config.json.  This will be used to determine if your config is out of date.

4. When config is out of date, old config is merged with new defaults.  Compare old config with the config-defaults.json.  Old settings take priority over defaults.  Any missing fields are auto populated with default.

5. A backup of your old config is generated and not deleted.  User can manually delete this file if they want.

If this sounds good I will code it up.

**Comment by Brazuca87** [comment]
That actually sounds awesome! :)
